src/com/owncloud/android/files/FileOperationsHelper.java

```diff
@@ -247,23 +247,59 @@ public void createFolder(String remotePath, boolean createFullPath) {
         mFileActivity.showLoadingDialog();
     }
 
+    /**
+     * Retry uploading a failed or cancelled upload with force.
+     */
+    public void retryUpload(UploadDbObject upload) {
+        Account account = mFileActivity.getAccount();
+        FileUploaderBinder uploaderBinder = mFileActivity.getFileUploaderBinder();
+        if (uploaderBinder != null) {
+            upload.removeAllUploadRestrictions(); //only this object, upload DB stays untouched.
+            uploaderBinder.retry(account, upload);            
+        }  else {
+            Log_OC.w(TAG, "uploaderBinder not set. Cannot remove " + upload.getOCFile());            
+        }
+    }
     
+    /**
+     * Remove upload from upload list.
+     */
+    public void removeUploadFromList(UploadDbObject upload) {
+        Account account = mFileActivity.getAccount();
+        FileUploaderBinder uploaderBinder = mFileActivity.getFileUploaderBinder();
+        if (uploaderBinder != null) {
+            uploaderBinder.remove(account, upload.getOCFile());            
+        }  else {
+            Log_OC.w(TAG, "uploaderBinder not set. Cannot remove " + upload.getOCFile());            
+        }
+    }
     public void cancelTransference(OCFile file) {
         Account account = mFileActivity.getAccount();
         FileDownloaderBinder downloaderBinder = mFileActivity.getFileDownloaderBinder();
-        FileUploaderBinder uploaderBinder =  mFileActivity.getFileUploaderBinder();
-        if (downloaderBinder != null && downloaderBinder.isDownloading(account, file)) {
-            // Remove etag for parent, if file is a keep_in_sync
-            if (file.keepInSync()) {
-               OCFile parent = mFileActivity.getStorageManager().getFileById(file.getParentId());
-               parent.setEtag("");
-               mFileActivity.getStorageManager().saveFile(parent);
+        FileUploaderBinder uploaderBinder = mFileActivity.getFileUploaderBinder();
+        if (downloaderBinder != null) {
+            if (downloaderBinder.isDownloading(account, file)) {
+                // Remove etag for parent, if file is a keep_in_sync
+                if (file.keepInSync()) {
+                    OCFile parent = mFileActivity.getStorageManager().getFileById(file.getParentId());
+                    parent.setEtag("");
+                    mFileActivity.getStorageManager().saveFile(parent);
+                }
+
+                downloaderBinder.cancel(account, file);
+            } else {
+                Log_OC.d(TAG, "Download for " + file + " not in progress. Cannot cancel " + file);
             }
-            
-            downloaderBinder.cancel(account, file);
-            
-        } else if (uploaderBinder != null && uploaderBinder.isUploading(account, file)) {
-            uploaderBinder.cancel(account, file);
+        } 
+        if (uploaderBinder != null) {
+            if (uploaderBinder.isUploading(account, file)) {
```

Comment:
This is fine, but I'm curious. Any specific reason (bug) to rewrite this part, or just tidying up?